### PR TITLE
DSLSchema Type for Validators and DataRefs

### PR DESCRIPTION
**Pull request by lexfm**
### Change Type (required)
Player DSLSchema type for Validators and DataRefs type-checking.

- Added `DSLSchema` type for setting up structure about the data types and validators used in the workspace when authoring DSL data schemas.
- Added `DataTypeReference` type to be used as the generic structure for the `DSLSchema` type.
- Created `DataTypeRefs` and  `ValidatorFunctionRefs` types to help generating the necessary data type reference and validator function objects type, so we can properly fill in `DataTypeReference` for the `DSLSchema`.
 - Implemented `getObjectReferences` helper function to help created the actual data type references based from the data type references type object, which can then be used when authoring DSL data schemas.

- [ ] `patch`
- [ ] `minor`
- [ ] `major`

<details>
  <summary>📦 Published PR as canary version: <code>0.4.2--canary.69.2028</code></summary>
  <br />

  Try this version out locally by upgrading relevant packages to 0.4.2--canary.69.2028
</details>

### CODE REVIEW

**Review by rafbcampos — APPROVED** [review]
LGTM

File: language/dsl/README.md [hunk 1]
```diff
@@ -7,3 +7,105 @@ The `@player-ui/dsl` package provides the primitives and build tools to create P
 Similar to React, Vue, or other component libraries, in order for users to take advantage of the JSX capabilities, they need a library of common components to pull from.
 
 As an asset author, you can create and share a library of `Components` for content authors to import and leverage.
+
```
Comment (KetanReddy) [inline language/dsl/README.md:10]: I don't think this documentation needs to live here as well as on the doc site.

**Review by KetanReddy — APPROVED** [review]

**Review by KetanReddy — APPROVED** [review]

**Pull request merged**